tests/unit_tests/operator_tests/Volume_operators_test.cpp

```diff
@@ -0,0 +1,298 @@
+        typename dealii::Triangulation<dim>::MeshSmoothing(
+            dealii::Triangulation<dim>::smoothing_on_refinement |
+            dealii::Triangulation<dim>::smoothing_on_coarsening));
+#endif
+        dealii::GridGenerator::hyper_cube (*grid, left, right, colorize);
+        grid->refine_global(igrid);
+
+        OPERATOR::OperatorBase<dim,nstate,real> operators(&all_parameters_new, poly_degree, poly_degree, poly_degree); 
+
+        const unsigned int n_dofs = nstate * pow(poly_degree+1,dim);
+       // dealii::QGaussLobatto<dim> vol_quad_GLL (poly_degree+1+overint);
+        dealii::QGaussLobatto<dim> vol_quad_GLL (poly_degree+1);
+        const std::vector<real> &quad_weights = vol_quad_GLL.get_weights ();
+#if 0
+pcout<<"GLL"<<std::endl;
+        for(unsigned int idof=0; idof<n_dofs; idof++){
+            for(unsigned int idof2=0; idof2<n_dofs; idof2++){
+                if(idof==idof2)
+                    pcout<<quad_weights[idof]<<" ";
+                else
+                    pcout<<0<<" ";
+            }
+    pcout<<std::endl;
+        }
+pcout<<"M+K"<<std::endl;
+        for(unsigned int idof=0; idof<n_dofs; idof++){
+            for(unsigned int idof2=0; idof2<n_dofs; idof2++){
+                if(std::abs(operators.local_mass[poly_degree][idof][idof2] +
+                            operators.local_K_operator[poly_degree][idof][idof2] )<1e-12)
+                    pcout<<0<<" ";
+                else
+                    pcout<<operators.local_mass[poly_degree][idof][idof2] +
+                            operators.local_K_operator[poly_degree][idof][idof2]<<" ";
+            }
+    pcout<<std::endl;
+        }
+#endif
+
+
+        for(unsigned int idof=0; idof<n_dofs; idof++){
+            for(unsigned int idof2=0; idof2<n_dofs; idof2++){
+                if(idof == idof2){
+                    const unsigned int ishape = operators.fe_collection_basis[poly_degree].system_to_component_index(idof).second;
+                    if(std::abs(quad_weights[ishape] - operators.local_mass[poly_degree][idof][idof2] -
+                            operators.local_K_operator[poly_degree][idof][idof2]) > 1e-12)
+                        M_K_HU = std::abs(quad_weights[idof] - operators.local_mass[poly_degree][idof][idof2] +
+                            operators.local_K_operator[poly_degree][idof][idof2]);
+                }
+                else{
+                    if(std::abs(operators.local_mass[poly_degree][idof][idof2] +
+                            operators.local_K_operator[poly_degree][idof][idof2]) > 1e-12)
+                        if(std::abs(operators.local_mass[poly_degree][idof][idof2] +
+                            operators.local_K_operator[poly_degree][idof][idof2]) > M_K_HU)
+                            M_K_HU = std::abs(operators.local_mass[poly_degree][idof][idof2] +
+                            operators.local_K_operator[poly_degree][idof][idof2]); 
+                }
+            }
+        }
+#if 0
```

Comment:
Feel free to delete those \#if 0 or comment blocks for a cleaner code if you are done debugging. Since you have commited them they can always be retrieved within Git's log by checkout out previous commits. If it's a work in progress, of course, please do keep them.

Reply:
Yes, I left those for now because it has all the covariant curl coded up. Was thinking of maybe adding it as an option? Like we could choose between invariant and conservative curl forms from the control file. If you'd like, we could open it as a topic/issue to be addressed in merge part 2?

Reply:
Yup! More to-dos to be added to issues

Reply:
Sorry I thought this was in a different place. Yes those if statements are needed either way. What this is doing is comparing that M+K_HU recovers exactly a collocated DG mass matrix on GLL nodes of degree p+1. So the if statements are comparing them and writing into M_K_HU the maximum error.